### PR TITLE
HASS 35 Tuning

### PR DESCRIPTION
**Pull request by john-chrosniak**
# PR Details
## Description

This PR includes updates to the config files for each model after tuning to improve pedestrian detection performance. Additionally, the script used to visualize detections was changed to only show pedestrian detections.

The following config changes were made during tuning:
1. The data augmentation setup initially in `pointpillar_newaugs.yaml` was copied over to both `second.yaml` and `centerpoint.yaml` to give each model an identical data augmentation setup during training.
2. The parameter `USE_ROAD_PLANE` in `kitti_dataset.yaml` was switched to `False`, as using road plane files has reportedly not improved performance by much and would require we create these files for each LiDAR scan in our dataset. A discussion on this can be found here: https://github.com/sshaoshuai/PointRCNN/issues/12
3. The `intensity` field was not used during training (turned off by removing it from `used_feature_list` in `kitti_dataset.yaml` since our dataset does not have intensity data. Thus, we don't want models to learn to rely on a feature during training that it will not have during testing.
4. The voxel size in `pointpillar_newaugs.yaml` was reduced from `0.16 m` to `0.08 m` since smaller voxels have been shown to improve pedestrian detection performance. The voxel size for other models was left at the default setting of `0.05 m` in `kitti_dataset.yaml`.
5. The number of epochs for each model was reduced from 80 to 25 since early stopping resulted in models generalizing better to our dataset.

## Related GitHub Issue

NA

## Related Jira Key

[HASS-35](https://usdot-carma.atlassian.net/browse/HASS-35)

## Motivation and Context

These updates were made to improve the pedestrian detection performance capability for each model.

## How Has This Been Tested?

These changes were used to generate the results in [this document](https://usdot-carma.atlassian.net/wiki/spaces/HS/pages/3256909825/LiDAR+Model+Tuning+for+Pedestrian+Detection+and+Overfitting+Mitigation).

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

[HASS-35]: https://usdot-carma.atlassian.net/browse/HASS-35?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tools/cfgs/dataset_configs/kitti_dataset.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ DATA_AUGMENTOR:
     DISABLE_AUG_LIST: ['placeholder']
     AUG_CONFIG_LIST:
         - NAME: gt_sampling
-          USE_ROAD_PLANE: True
+          USE_ROAD_PLANE: False
```
Comment (JonSmet) [inline tools/cfgs/dataset_configs/kitti_dataset.yaml:23]: Could we add to the [Confluence page](https://usdot-carma.atlassian.net/wiki/spaces/HS/pages/3256909825/LiDAR+Model+Tuning+for+Pedestrian+Detection+and+Overfitting+Mitigation) that we adjusted this setting? Is this removing ground point cloud points?

File: tools/cfgs/kitti_models/pointpillar_newaugs.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 CLASS_NAMES: ['Car', 'Pedestrian', 'Cyclist']
 
-DATA_CONFIG: 
+DATA_CONFIG:
```
Comment (JonSmet) [inline tools/cfgs/kitti_models/pointpillar_newaugs.yaml:3]: Is it worth updating the Confluence page to specify the model config files that were used, since there are multiple PointPillar model config files for example (pointpillar.yaml, pointpillar_newaugs.yaml, pointpillar_pyramid_aug.yaml).

File: tools/cfgs/kitti_models/PartA2_free.yaml [hunk 3]
```diff
@@ -2,6 +2,53 @@ CLASS_NAMES: ['Car', 'Pedestrian', 'Cyclist']
 
 DATA_CONFIG:
     _BASE_CONFIG_: cfgs/dataset_configs/kitti_dataset.yaml
+    DATA_AUGMENTOR:
```
Comment (JonSmet) [inline tools/cfgs/kitti_models/PartA2_free.yaml:5]: Just checking, this config file has been edited but it wasn't included in the results documented on https://usdot-carma.atlassian.net/wiki/spaces/HS/pages/3256909825/LiDAR+Model+Tuning+for+Pedestrian+Detection+and+Overfitting+Mitigation ?

File: tools/cfgs/kitti_models/centerpoint.yaml [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+            MULTI_CLASSES_NMS: False
+            NMS_TYPE: nms_gpu
+            NMS_THRESH: 0.01
+            NMS_PRE_MAXSIZE: 4096
+            NMS_POST_MAXSIZE: 500
+
+
+OPTIMIZATION:
+    BATCH_SIZE_PER_GPU: 4
+    NUM_EPOCHS: 80
```
Comment (JonSmet) [inline tools/cfgs/kitti_models/centerpoint.yaml:134]: Placing this comment here, but it is more general to the PR - the Confluence page mentions the models were trained for 25 epochs, is this reflected in any of these model configs?

File: tools/cfgs/kitti_models/pointpillar_newaugs.yaml [hunk 5]
```diff
@@ -150,7 +150,7 @@ MODEL:
 
     POST_PROCESSING:
         RECALL_THRESH_LIST: [0.3, 0.5, 0.7]
-        SCORE_THRESH: 0.1
+        SCORE_THRESH: 0.4
```
Comment (JonSmet) [inline tools/cfgs/kitti_models/pointpillar_newaugs.yaml:153]: The other model config files have `SCORE_THRESH` set to 0.1, is this change intended? Or was this manually adjusted for each iteration during testing (i.e. 0.1, then 0.2, then 0.3, then 0.4)?

File: tools/cfgs/kitti_models/centerpoint.yaml [hunk 6]
```diff
@@ -0,0 +1,151 @@
+CLASS_NAMES: ['Car', 'Pedestrian', 'Cyclist']
```
Comment (JonSmet) [inline tools/cfgs/kitti_models/centerpoint.yaml:1]: This is a more general comment on the PR, but due to the nature of our larger team not having a deep understanding of OpenPCDet, could the PR description be updated with more information on these changes at a high level in order to help a new user follow these changes? (i.e. "the `DATA_AUGMENTOR` configuration was copied from pointpillar_newaugs.yaml and ported to the other configs in order to inject noise into the dataset for training", note that this may not be accurate, but just an example)

**Review by JonSmet — CHANGES_REQUESTED** [review]

File: tools/cfgs/dataset_configs/kitti_dataset.yaml (hunk 1 above)
Comment (john-chrosniak) [reply tools/cfgs/dataset_configs/kitti_dataset.yaml:23]: This refers to whether or not we should use road plane files, which are points that describe a plane that roughly fits the ground plane in each point cloud. These are just used for data augmentation and have been noted to not significantly improve performance, plus would require that we generate these files for our own dataset, so I have it disabled. I can mention this in the Confluence page after merging

More details on road plane files: https://github.com/sshaoshuai/PointRCNN/issues/12

File: tools/cfgs/kitti_models/pointpillar_newaugs.yaml (hunk 2 above)
Comment (john-chrosniak) [reply tools/cfgs/kitti_models/pointpillar_newaugs.yaml:3]: Yeah that's a good point, I'll add that in

File: tools/cfgs/kitti_models/PartA2_free.yaml (hunk 3 above)
Comment (john-chrosniak) [reply tools/cfgs/kitti_models/PartA2_free.yaml:5]: Yes I looked into this algorithm initially but it didn't give good results on the first experiment I ran so I didn't do any additional experiments. I'll get rid of this change since it's not needed.

File: tools/cfgs/kitti_models/centerpoint.yaml (hunk 4 above)
Comment (john-chrosniak) [reply tools/cfgs/kitti_models/centerpoint.yaml:134]: Not currently but I'll update them to reflect this.

File: tools/cfgs/kitti_models/pointpillar_newaugs.yaml (hunk 5 above)
Comment (john-chrosniak) [reply tools/cfgs/kitti_models/pointpillar_newaugs.yaml:153]: This was manually adjusted during testing. I'll set them all to 0.1 for now since that was the default setting and mention in Confluence that this is the parameter to adjust.
